### DISCUSSION TITLE
touchscreen support [wayland]

### DISCUSSION THREAD

**Discussion by usernameavailabl3** [qtile/qtile#3915] [Q&A]
Hello, my laptops touchscreen (wacom) doesn't seem to work under wayland. According to libinput (debug-events) it works, as in, it detects when I touch my screen, but my cursor isn't moving. Is this an issue with qtile or do I just have to configure it somewhere?
note: `qtile cmd-obj -o core -f get_inputs` doesn't list it.

**Comment by m-col** [comment]
Qtile doesn't support this, but I believe it would be relatively straight forward to support. I think I could put something together over the next week or so but I don't have a touchscreen to test it with, so it may need some collaboration.

**Reply by usernameavailabl3** [reply]
thanks, I wouldn't mind testing it

**Reply by m-col** [reply]
I tried implementing it here: https://github.com/qtile/qtile/pull/3931

Would you mind giving it a go and seeing it works? Hopefully the cursor should move and 'clicking' should work as expected.

**Reply by usernameavailabl3** [reply]
Just tried it, doesn't seem to work, but `qtile cmd-obj -o core -f get_inputs` now detects wacom finger.

**Comment by nerothank** [comment]
Hi @m-col 

Sorry to ping you from such an old topic.

I just stumbled upon this when looking for ideas on how to hack touch support into qtile. I already had quite similar code as the one you wrote two years ago (should have asked the internet before doing it by myself ;) ).

However, I am stuck at the same place as you and the requestor were two years ago. Touch is detected by `libinput debug-events`, yet neither does the cursor move nor are the "clicks" being registered as such by qtile.

I am quite adapt with python, yet a novice regarding libinput (or anything hardware/compositor related, really).
Would you happen to have any pointers as to where I could start looking into resolving this?

**Reply by nerothank** [reply]
Quick update: I was able to get basic touch functionality working. Basically all that was missing were the listeners for the touch events.
The code I wrote is pretty hacky, though. So I will not create a PR for the moment.
If anyone is interested: https://github.com/qtile/qtile/compare/master...ttrafelet:qtile:master
